Cairo Desktop/Cairo Desktop/DynamicDesktopNavigationManager.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace CairoDesktop
+{
+    internal class DynamicDesktopNavigationManager
+    {
+        private List<string> list;
+        private int currentIndex;
+        private IReadOnlyList<string> readOnlyList;
+
+        internal Action<string> Navigating;
+
+        public DynamicDesktopNavigationManager()
+        {
+            list = new List<string>();
+            currentIndex = 0;
+        }
+
+        public void Clear()
+        {
+            list.Clear();
+            currentIndex = 0;
```

Comment:
There seems to be a problem with clear history not seeding the current directory. In your home desktop folder, add a folder if there isnt one there. Then open it, then go back. Then clear history and then open it again and try to go back.. it wont go back